src/EnergyPlus/OutsideEnergySources.cc

```diff
@@ -428,21 +472,48 @@ void OutsideEnergySourceSpecs::calculate(EnergyPlusData &state, bool runFlag, Re
+            this->EnergyType == DataPlant::PlantEquipmentType::PurchHotWater) {
+            Real64 const Cp = FluidProperties::GetSpecificHeatGlycol(
+                state, state.dataPlnt->PlantLoop(LoopNum).FluidName, this->InletTemp, state.dataPlnt->PlantLoop(LoopNum).FluidIndex, RoutineName);
+            this->OutletTemp = (MyLoad + this->MassFlowRate * Cp * this->InletTemp) / (this->MassFlowRate * Cp);
+            // apply loop limits on temperature result to keep in check
+            if (this->OutletTemp < LoopMinTemp) {
+                this->OutletTemp = max(this->OutletTemp, LoopMinTemp);
+                MyLoad = this->MassFlowRate * Cp * (this->OutletTemp - this->InletTemp);
+            }
+            if (this->OutletTemp > LoopMaxTemp) {
+                this->OutletTemp = min(this->OutletTemp, LoopMaxTemp);
+                MyLoad = this->MassFlowRate * Cp * (this->OutletTemp - this->InletTemp);
+            }
+        } else if (this->EnergyType == DataPlant::PlantEquipmentType::PurchSteam) { // determine mass flow rate based on MyLoad
+            Real64 EnthSteamInDry = FluidProperties::GetSatEnthalpyRefrig(state,
+                                                                          state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidName,
+                                                                          this->InletTemp,
+                                                                          1.0,
+                                                                          state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidIndex,
+                                                                          RoutineName);
+            Real64 EnthSteamOutWet = FluidProperties::GetSatEnthalpyRefrig(state,
+                                                                           state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidName,
+                                                                           this->InletTemp,
+                                                                           0.0,
+                                                                           state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidIndex,
+                                                                           RoutineName);
+            Real64 LatentHeatSteam = EnthSteamInDry - EnthSteamOutWet;
+            Real64 SteamMdot = MyLoad / LatentHeatSteam;
+            PlantUtilities::SetComponentFlowRate(state, SteamMdot, this->InletNodeNum, this->OutletNodeNum, this->plantLoc);
+            this->MassFlowRate = state.dataLoopNodes->Node(this->OutletNodeNum).MassFlowRate;
+            this->OutletTemp = state.dataLoopNodes->Node(state.dataPlnt->PlantLoop(this->plantLoc.loopNum).TempSetPointNodeNum).TempSetPoint;
+            // Like the assumption in Boiler:Steam, assume that it can meet the steam loop setpoint
+            this->OutletSteamQuality = 1.0;
```

Comment:
Is this right? Quality = 1? The mass flow rate is calculated to bring the steam all the way to EnthSteamOutWet, which would be a quality of 0. See line 505 above, and 508 and 509. And at line 520, Quality would be 1.

Reply:
I believe that the outlet steam quality of the district heating steam is 1. (And the inlet steam quality of the district heating steam is 0.) According to EnergyPlus' assumptions for the steam system, the steam loop in EnergyPlus is designed to operate under saturated conditions, with no superheated vapor. Also, a `Boiler:Steam` generates steam with a quality of 1 each time. Similarly, the `DistrictHeatingSteam` component is expected to provide steam with a quality of 1 to the demand side of the steam loop.  `EnthSteamInDry`, `EnthSteamOutWet`, and `LatentHeatSteam` are interim variables that are used to calculate the `SteamMdot`.

Reply:
Yes, the quality of the steam leaving the district heating plant = 1. Bu this calculation is not the quality of the steam from the district heating plant. It's the quality of the steam after it is used to meet a load for outside energy sources, specifically MyLoad.

If I look at the call for EnthSteamOutWet I see Quality = 0 in argument #4. While Quality = 1 for EnthSteamInDry. That makes sense, right? Then:

    LatentHeatSteam = EnthSteamInDry - EnthSteamOutWet;, and
    SteamMdot = MyLoad / LatentHeatSteam;

So the model is setting the steam mass flow rate to pull all the latent heat out of the fluid. If that happens, and it will if Mdot = SteamMdot, which may not be the case after the call to SetComponentFlowRate, then all the latent heat was removed. That means the fluid quality has changed. It may not get to 0 (if the plant limits the steam flow rate), but it is certainly not 1. And quality will usually be 0 at this point (if flow rate is not limited).

    Real64 EnthSteamOutWet = FluidProperties::GetSatEnthalpyRefrig(state,
                                                                           state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidName,
                                                                           this->InletTemp,
                                                                           0.0,
                                                                           state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidIndex,
                                                                           RoutineName);

Reply:
I got your point. Thanks for the explanation. I made the component member variable `OutletSteamQuality` and regarded it as the quality of steam leaving the district heating steam plant. I'll fix the code as per your insight. Or, I could remove this member variable since it's pretty much a useless variable.

Reply:
I think just leave the variable for now, just make sure it reports the correct value. Quality = 1 for steam, quaulity = 0 for water at 100 C. If the component is on quality out = 0, if component is off quality out = 1,

Reply:
I don't even know what quality means in the context of steam in E+. I mean it's steam, 0 means it's hot water. What does 0.5 mean? in an open system. If it means the fluid is half way to being water then I am totally lost. It;s an enthalpy question.

Reply:
The assumption that it cannot allow superheated vapor makes everything simple but a little awkward. It would be much better if EnergyPlus's steam system could ultimately be upgraded to an enthalpy-based system, enabling the use of superheated vapor (someday, if funding allows…).